src/psyclone/psyir/nodes/loop.py

```diff
@@ -102,6 +102,7 @@ def __init__(self, variable=None, annotations=None, **kwargs):
         self._variable = None
         if variable is not None:
             self.variable = variable
+        self._explicitly_local_symbols = set()
```

Comment:
I started the branch calling it "_explicitly_private_symbols". Would that be a better name? In fact in the short term it will only populate the "private" clauses of OpenMP or OpenACC.

The reason I changed it is because Fortran in "do concurrent" calls this "local"
```
R1130 locality-spec is LOCAL ( variable-name-list )
                                or LOCAL_INIT ( variable-name-list )
                                or REDUCE ( reduce-operation : variable-name-list )
                                or SHARED ( variable-name-list )
                                or DEFAULT ( NONE )
```
supposedly because "private" is already used for the visibility, and it the past I regretted some PSyIR concepts diverging from the Fortran notation. But maybe private is clearer in this case? What do you think?

There was another alternative without storing this set at all, and instead bringing the symbol to the inner loop_body schedule symbol table. This will imply that its scope is local without any property needed. But this required more changes elsewhere, but I may come back to this as I find it an elegant solution. I added this as a comment for now.

Reply:
I too like the scoping idea but I can see that will need more work. I also agree that the new name is more intuitive.